plugins/chat/app/services/chat/create_direct_message_channel.rb

```diff
@@ -58,17 +61,26 @@ def actor_allows_dms(user_comm_screener:, **)
       !user_comm_screener.actor_disallowing_all_pms?
     end
 
-    def fetch_or_create_direct_message(target_users:, **)
-      Chat::DirectMessage.for_user_ids(target_users.map(&:id)) ||
-        Chat::DirectMessage.create(user_ids: target_users.map(&:id))
+    def fetch_or_create_direct_message(target_users:, contract:, **)
+      ids = target_users.map(&:id)
+
+      if ids.size > 2 || contract.name.present?
+        ::Chat::DirectMessage.create(user_ids: ids, group: true)
+      else
+        ::Chat::DirectMessage.for_user_ids(ids) || ::Chat::DirectMessage.create(user_ids: ids)
+      end
```

Comment:
This is an important change and mostly how it works.  A channel created for more than two users and/or with a name becomes a group channel.

Otherwise it's a regular direct message channel, or we reuse an existing one for the same user_ids if it exists